src/Deposit.js

```diff
@@ -568,9 +626,20 @@ export default class Deposit {
       parseInt(requiredConfirmations)
     )
     proofArgs.unshift(this.address)
-    const transaction = await this.factory.vendingMachineContract.methods
-      .unqualifiedDepositToTbtc(...proofArgs)
-      .send()
+
+    // Use approveAndCall pattern to execute VendingMachine.unqualifiedDepositToTbtc.
+    const unqualifiedDepositToTbtcCall = this.factory.vendingMachineContract.methods
+      .unqualifiedDepositToTbtc(this.address, ...proofArgs)
+      .encodeABI()
+
+    const transaction = await EthereumHelpers.sendSafely(
+      this.factory.tokenContract.methods.approveAndCall(
```

Comment:
Are we approving transfer on the token contract or the deposit token contract?

Reply:
Just swapped from using git stashes to 'wip' branches. This must've slipped through, woops.